src/Platform/Microsoft.Testing.Extensions.VSTestBridge/ObjectModel/ObjectModelConverters.cs

```diff
@@ -90,38 +89,29 @@ private static void CopyVSTestProperties(IEnumerable<TestProperty> testPropertie
                 {
                     testNode.Properties.Add(new SerializableNamedArrayStringProperty("vstest.TestCase.Hierarchy", testCaseHierarchy));
                 }
+            }
 
-                // ID is defined on TraitCollection but is internal so again we copy the string here.
-                else if (property.Id == "TestObject.Traits"
-                    && getPropertyValue(property) is KeyValuePair<string, string>[] traits && traits.Length > 0)
+            // ID is defined on TraitCollection but is internal so again we copy the string here.
+            if (property.Id == "TestObject.Traits"
+                && getPropertyValue(property) is KeyValuePair<string, string>[] traits && traits.Length > 0)
+            {
+                foreach (KeyValuePair<string, string> trait in traits)
                 {
-#pragma warning disable SA1010 // Opening square brackets should be spaced correctly
-                    testNodeTraits ??= [];
-#pragma warning restore SA1010 // Opening square brackets should be spaced correctly
-                    testNodeTraits.AddRange(traits);
+                    testNode.Properties.Add(new TestMetadataProperty(trait.Key, trait.Value));
```

Comment:
Doing this it looks like we will not be able to separate traits from other test metadata anymore? This might be unexpected for VS. Have you tested what we see in TE after this change in the traits dropdown / selection ?

Reply:
Looking at the json below in tests `"traits":[{"testmetadata-key":"testmetadata-value"}]`

we see testMetadataProperty  collected here in the trait key. Looks like we would have a separate object for each key, so we could change to:
```
"traits":[
    {
        "testmetadata-key":"testmetadata-value",
        "test-metadata-type": "platform-metadata"
    },
]
```

But I am unsure it that is needed.

Reply:
@nohwnd I'm not following. TestMetadataProperty and traits are the same thing. The only thing that we should have on MTP is just `TestMetadataProperty`.

Reply:
I thought that what we shown in VS was the data produced by Traits / Category however each framework calls that attribute. And in MTP traits is bigger group of all possible properties and test metadata you can put on a test. In TATF there was a KVP Property on the test. So if I wanted to add a description to a test I would add "description"="this is my tests and I like it described", but that is not a category of a test that should show up in TE (in my opinion).

Reply:
Ok. I will leave it up to your judgement :)